### PR TITLE
Refactor domain issue and avoid injection

### LINKED ISSUE #89: Refactor domain issue and avoid injection
### Motivation

We need to refactor the `Issue` in our domain to avoid injecting it everywhere in the `IssueTracker`, simplifying the flow and the code.

### PR DESCRIPTION
**Pull request by hielfx**
Closes #89

### CODE REVIEW

File: internal/branches/interactive.go [hunk 1]
```diff
@@ -11,28 +11,31 @@ import (
...
+	// issueType := issueTracker.GetIssueType(issue)
+	issueType := issue.Type()
 	branchType := issueType.String()
 
-	issueID := issueTracker.FormatIssueId(issue.ID)
+	// formattedID := issueTracker.FormatIssueId(issue.ID)
+	formattedID := issue.FormatID()
 
-	issueSlug := parseIssueContext(issue.Title)
+	issueSlug := parseIssueContext(issue.Title())
```
Comment (ismaelgonval) [inline internal/branches/interactive.go:27]: suggestion: parseIssueContext is a bit of a strange name for its functionality. It looks like is normalizing branch name.

File: internal/domain/issue.go [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
-	Id          string
-	Name        string
-	Description string
+type Issue interface {
+	FormatID() string
+	ID() string
+	Title() string
+	Body() string
+	URL() string
+	TypeLabel() string
```
Comment (ismaelgonval) [inline internal/domain/issue.go:29]: question: only type labels here?

File: internal/issue_trackers/github/github.go [hunk 3]
```diff
@@ -74,26 +72,16 @@ func (g *Github) GetIssue(identifier string) (issue domain.Issue, err error) {
...
-		Url:          result.Url,
-		Labels:       labels,
-		IssueTracker: domain.IssueTrackerTypeGithub,
+	return Issue{
+		id:           strconv.FormatInt(result.Number, 10),
+		title:        result.Title,
+		body:         result.Body,
+		url:          result.Url,
+		labels:       labels,
+		labelsConfig: g.cfg.IssueLabels,
```
Comment (ismaelgonval) [inline internal/issue_trackers/github/github.go:81]: thought: config embed in this model looks really strange.

File: internal/use_cases/create_pull_request.go [hunk 4]
```diff
@@ -55,12 +55,13 @@ func (cpr CreatePullRequest) Execute() error {
 
 	// 1. FLAG ISSUE IS USED
 	if issueID != "" {
-		issueTracker, err := cpr.IssueTrackerProvider.GetIssueTracker(issueID)
+		_issue, err := cpr.IssueTrackerProvider.GetIssue(issueID)
```
Comment (ismaelgonval) [inline internal/use_cases/create_pull_request.go:58]: question: why leading underscore?

File: internal/issue_trackers/github/github.go (hunk 3 above)
Comment (hielfx) [reply internal/issue_trackers/github/github.go:81]: Inject single label on issue creation

File: internal/issue_trackers/github/github_test.go [hunk 5]
```diff
@@ -113,72 +165,66 @@ func Test_CheckConfiguration(t *testing.T) {
...
+// 			want:  issue_types.Unknown,
+// 		},
+// 	} {
+// 		tc := tc
+// 		t.Run(tc.name, func(t *testing.T) {
+// 			got := tc.issue.Type()
+// 			assert.Equal(t, tc.want, got)
+// 		})
+// 	}
+// }
```
Comment (ismaelgonval) [inline internal/issue_trackers/github/github_test.go:227]: issue: remove commented code

File: internal/issue_trackers/github/github_test.go [hunk 6]
```diff
@@ -211,92 +257,79 @@ func Test_IdentifyIssue(t *testing.T) {
...
+// 			want:  "",
+// 		},
+// 	} {
+// 		tc := tc
+// 		t.Run(tc.name, func(t *testing.T) {
+// 			got := tc.issue.TypeLabel()
+// 			assert.Equal(t, tc.want, got)
+// 		})
+// 	}
+// }
```
Comment (ismaelgonval) [inline internal/issue_trackers/github/github_test.go:335]: issue: remove commented code

File: internal/issue_trackers/jira/jira_test.go [hunk 7]
```diff
@@ -1,171 +1,289 @@
...
+// 			want:  "",
+// 		},
+// 	} {
+// 		tc := tc
+// 		t.Run(tc.name, func(t *testing.T) {
+// 			got := tc.issue.TypeLabel()
+// 			assert.Equal(t, tc.want, got)
+// 		})
+// 	}
+// }
```
Comment (ismaelgonval) [inline internal/issue_trackers/jira/jira_test.go:289]: issue: remove commented code

File: internal/use_cases/create_pull_request_test.go [hunk 8]
```diff
@@ -417,17 +398,26 @@ func (s *CreateJiraPullRequestExecutionTestSuite) SetupSubTest() {
...
-	s.issueTracker = domainFakes.NewFakeIssueTracker()
-	s.issueTracker.IssueTrackerType = domain.IssueTrackerTypeJira
-	s.issueTracker.AddIssue("PROJECTKEY-1", issue_types.Feature)
-	s.issueTracker.AddIssue("PROJECTKEY-3", issue_types.Documentation)
-	s.issueTracker.AddIssue("PROJECTKEY-6", issue_types.Refactoring)
+	// s.issueTracker = domainFakes.NewFakeIssueTracker()
+	// s.issueTracker.IssueTrackerType = domain.IssueTrackerTypeJira
+	// s.issueTracker.AddIssue("PROJECTKEY-1", issue_types.Feature)
+	// s.issueTracker.AddIssue("PROJECTKEY-3", issue_types.Documentation)
+	// s.issueTracker.AddIssue("PROJECTKEY-6", issue_types.Refactoring)
```
Comment (ismaelgonval) [inline internal/use_cases/create_pull_request_test.go:405]: issue: remove commented code

**Review by ismaelgonval — APPROVED** [review]

**Pull request merged**